eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/ResourceCollectionClientProvider.cs

```diff
@@ -233,7 +233,7 @@ protected override bool SkipMethodParameter(ParameterProvider parameter)
             return ContextualParameters.Take(ContextualParameters.Count - 1).Any(p => p == parameter.Name);
         }
 
-        protected override MethodBodyStatement BuildReturnStatements(ValueExpression responseVariable, MethodSignature signature)
+        protected override IReadOnlyList<MethodBodyStatement> BuildReturnStatements(ValueExpression responseVariable, MethodSignature signature)
         {
             if (signature.Name == "GetIfExists" || signature.Name == "GetIfExistsAsync")
```

Comment:
Can we refine BuildOperationMethodCore further?
It's odd to apply parts of the method body differently depending on method name.
Can we abstract something like ResourceOperationMethodProvider? Then we can have a different instance of it  for each  method.